### DISCUSSION TITLE
random recompile and page reload in development

### DISCUSSION THREAD

**Discussion by macrozone** [vercel/next.js#24104] [Help]
i have a weird problem since some versions of nextjs.

My app reloads every 30 seconds or so. I see that the server triggers a "compiling..." and then the client will receive an event to reload the page.

this is really annoying, but I have no idea why it happens.

The appication uses a custom server, so maybe its related to nodemon? Anyone has an idea?

it it possible log why nextjs rebuild the app?

**Comment by bopfer** [comment]
Also seeing this issue.  Reloads about every minute for me.  Did you figure anything out?

**Reply by bopfer** [reply]
Some followup... I am still not really sure _why_ the reloading occurs, but adding this section to my `next.config.js` file has stopped it from happening:

```js
onDemandEntries: {
  maxInactiveAge: 60 * 60 * 1000,
  pagesBufferLength: 20,
},
```
